File: permissionsdispatcher-processor/src/main/java/permissions/dispatcher/processor/ConstantsProvider.java
package permissions.dispatcher.processor;

import com.squareup.javapoet.ClassName;

interface ConstantsProvider {

    ClassName ACTIVITY_COMPAT = ClassName.get("android.support.v4.app", "ActivityCompat");
    ClassName PERMISSION_UTILS = ClassName.get("permissions.dispatcher", "PermissionUtils");
    String CLASS_SUFFIX = "PermissionsDispatcher";
    String METHOD_SUFFIX = "WithCheck";
    String REQUEST_CODE_PREFIX = "REQUEST_";
    String PERMISSION_PREFIX = "PERMISSION_";

}

File: permissionsdispatcher-processor/src/main/kotlin/permissions/dispatcher/processor/ConstantsProvider.kt
package permissions.dispatcher.processor

import com.squareup.javapoet.ClassName

interface ConstantsProvider {
    companion object {

        public val ACTIVITY_COMPAT: ClassName = ClassName.get("android.support.v4.app", "ActivityCompat")
        public val PERMISSION_UTILS: ClassName = ClassName.get("permissions.dispatcher", "PermissionUtils")
        public val CLASS_SUFFIX: String = "PermissionsDispatcher"
        public val METHOD_SUFFIX: String = "WithCheck"
        public val REQUEST_CODE_PREFIX: String = "REQUEST_"
        public val PERMISSION_PREFIX: String = "PERMISSION_"
    }

}
